File: api/src/main/java/com/makingiants/today/api/repository/history/HistoryRepository.java
package com.makingiants.today.api.repository.history;

import com.makingiants.today.api.Api;
import com.makingiants.today.api.repository.history.pojo.Event;
import java.util.List;
import rx.Observable;

public class HistoryRepository {
  private HistoryService mHistoryService;

  public HistoryRepository() {
    mHistoryService = Api.create(HistoryService.class);
  }

  public Observable<List<Event>> get(int day, int month) {
    String formattedMoth = String.format("%02d", ++month);
    return mHistoryService.get(day, formattedMoth);
  }
}

File: api/src/main/kotlin/com/makingiants/today/api/repository/history/HistoryRepository.kt
package com.makingiants.today.api.repository.history

import com.makingiants.today.api.Api
import com.makingiants.today.api.repository.history.pojo.Event
import io.reactivex.Observable

open class HistoryRepository() {
  private val mHistoryService = Api.create(HistoryService::class.java)

  open fun get(day: Int, month: Int): Observable<List<Event>> =
      mHistoryService?.get(day, month) ?: Observable.empty()

}
